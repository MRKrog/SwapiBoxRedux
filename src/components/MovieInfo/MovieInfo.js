import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

const MovieInfo = ({ movieData }) => {
  return (
      <div className="MovieInfo">
        <div className="fade"></div>
        <section className="star-wars">
          <div className="crawl">
            <div className="title">
              <p>{movieData.title}</p>
              <h1>{movieData.release_date}</h1>
            </div>
            <p>{movieData.opening_crawl}</p>
          </div>
        </section>
        <NavLink to="/Explore">
          <button className="startBtn">Explore Movie</button>
        </NavLink>
      </div>
    )
}

MovieInfo.propTypes = {
  movieData: PropTypes.object
}

export default MovieInfo;
