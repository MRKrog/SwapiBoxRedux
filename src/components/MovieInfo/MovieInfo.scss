.MovieInfo {
  width: 100%;
  height: 100vh;
  background: #000;
  overflow: hidden;

  .fade {
    position: relative;
    width: 100%;
    min-height: 60vh;
    top: -25px;
    background-image: linear-gradient(0deg, transparent, black 75%);
    z-index: 1;
  }

  .star-wars {
    display: flex;
    justify-content: center;
    position: relative;
    height: 800px;
    color: #feda4a;
    font-family: 'Pathway Gothic One', sans-serif;
    font-size: 500%;
    font-weight: 600;
    letter-spacing: 4px;
    line-height: 150%;
    perspective: 400px;
    text-align: justify;
  }

  .crawl {
    position: relative;
    top: 9999px;
    transform-origin: 50% 100%;
    animation: crawl 60s linear;
  }

  .crawl > .title {
    font-size: 90%;
    text-align: center;
  }

  .crawl > .title h1 {
    margin: 0 0 100px;
    text-transform: uppercase;
  }

  @keyframes crawl {
    0% {
      top: 0;
      transform: rotateX(20deg)  translateZ(0);
    }
    100% {
      top: -6000px;
      transform: rotateX(25deg) translateZ(-2500px);
    }
  }

  .startBtn {
    position: absolute;
    color: #000000;
    bottom: 2%;
    z-index: 9999;
    font-size: .8em;
    line-height: 1;
    padding: .5em 1em;
    border-radius: 5px;
    border: none;
    outline: none;
    text-transform: uppercase;
    background: #D7C078;
    font-weight: bold;
    right: 2%;
    cursor: pointer;
    opacity: 1;
    transition: all .5s ease;

    &:hover {
      opacity: .7
    }
  }

}
