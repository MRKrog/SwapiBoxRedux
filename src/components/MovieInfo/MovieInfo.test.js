import React from 'react';
import MovieInfo from './MovieInfo';
import { shallow } from 'enzyme';

const handleUser = jest.fn();

describe('MovieInfo', () => {
  let wrapper;
  let movieMockData;

  beforeEach(() => {
    movieMockData = {
      title: "Revenge of the Sith",
      episode_id: 3,
      opening_crawl: "War! The Republic is crumbling nunder attacks by the ruthless Sith Lord",
      director: "George Lucas",
      producer: "Rick McCallum",
      release_date: "2005-05-19"
    }
    wrapper = shallow( <MovieInfo movieData={movieMockData} /> )
  })


  it("should match snapshot when all data is passed correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

})
