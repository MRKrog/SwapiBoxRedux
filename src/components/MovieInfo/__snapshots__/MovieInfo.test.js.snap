// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfo should match snapshot when all data is passed correctly 1`] = `
<div
  className="MovieInfo"
>
  <div
    className="fade"
  />
  <section
    className="star-wars"
  >
    <div
      className="crawl"
    >
      <div
        className="title"
      >
        <p>
          Revenge of the Sith
        </p>
        <h1>
          2005-05-19
        </h1>
      </div>
      <p>
        War! The Republic is crumbling nunder attacks by the ruthless Sith Lord
      </p>
    </div>
  </section>
  <NavLink
    to="/Explore"
  >
    <button
      className="startBtn"
    >
      Explore Movie
    </button>
  </NavLink>
</div>
`;
